assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetGA4/indexEntityDashboard.stories.js

```diff
@@ -0,0 +1,269 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	createTestRegistry,
+	provideModules,
+	provideSiteInfo,
+	WithTestRegistry,
+} from '../../../../../../../tests/js/utils';
+import {
+	getAnalytics4MockResponse,
+	provideAnalytics4MockReport,
+} from '../../../../analytics-4/utils/data-mock';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import { getWidgetComponentProps } from '../../../../../googlesitekit/widgets/util';
+import { MODULES_ANALYTICS_4 } from '../../../../analytics-4/datastore/constants';
+import { replaceValuesInAnalytics4ReportWithZeroData } from '../../../../../../../.storybook/utils/zeroReports';
+import DashboardAllTrafficWidgetGA4 from '.';
+
+function limitResponseToSingleDate( analyticsResponse ) {
+	const findFirstDateRangeRow = ( dateRange ) =>
+		analyticsResponse.rows.find(
+			( { dimensionValues } ) => dimensionValues[ 1 ].value === dateRange
+		);
+
+	return {
+		...analyticsResponse,
+		rows: [
+			findFirstDateRangeRow( 'date_range_0' ),
+			findFirstDateRangeRow( 'date_range_1' ),
+		],
+	};
+}
+
+const widgetComponentProps = getWidgetComponentProps(
+	'analyticsAllTraffic-widget'
+);
+
+const Template = () => (
+	<DashboardAllTrafficWidgetGA4 { ...widgetComponentProps } />
+);
+
+const baseAllTrafficOptions = {
+	startDate: '2020-12-09',
+	endDate: '2021-01-05',
+	compareStartDate: '2020-11-11',
+	compareEndDate: '2020-12-08',
+	metrics: [
+		{
+			name: 'totalUsers',
+		},
+	],
+};
+
+const allTrafficReportOptions = [
+	{
+		...baseAllTrafficOptions,
+		dimensions: [ 'sessionDefaultChannelGrouping' ],
+		orderby: {
+			fieldName: 'totalUsers',
+			sortOrder: 'DESCENDING',
+		},
+		limit: 6,
+		url: 'https://www.elasticpress.io/features/',
+	},
+	{
+		...baseAllTrafficOptions,
+		dimensions: [ 'country' ],
+		orderby: {
+			fieldName: 'totalUsers',
+			sortOrder: 'DESCENDING',
+		},
+		limit: 6,
+		url: 'https://www.elasticpress.io/features/',
+	},
+	{
+		...baseAllTrafficOptions,
+		dimensions: [ 'deviceCategory' ],
+		orderby: {
+			fieldName: 'totalUsers',
+			sortOrder: 'DESCENDING',
+		},
+		limit: 6,
+		url: 'https://www.elasticpress.io/features/',
+	},
+	{
+		...baseAllTrafficOptions,
+		url: 'https://www.elasticpress.io/features/',
+	},
+	{
+		startDate: '2020-12-09',
+		endDate: '2021-01-05',
+		dimensions: [ 'date' ],
+		metrics: [
+			{
+				name: 'totalUsers',
+			},
+		],
+		url: 'https://www.elasticpress.io/features/',
+	},
+];
+
+export const EntityDashboardLoaded = Template.bind( {} );
+EntityDashboardLoaded.storyName = 'Loaded';
+EntityDashboardLoaded.args = {
+	setupRegistry: ( registry ) => {
+		allTrafficReportOptions.forEach( ( options ) => {
+			provideAnalytics4MockReport( registry, options );
+		} );
+	},
+};
+EntityDashboardLoaded.scenario = {
```

Comment:
It is debatable whether it's worth keeping the scenarios for these Entity Dashboard stories, as they are visually identical to the Main Dashboard stories and it's fairly unlikely the Entity Dashboard ones would fail if the Main Dashboard are passing... :thinking:

Reply:
If they aren't providing any real value I'm fine with removing them too, but if the logic in the component to render them is at all different I'd say it's nice insurance keeping them around, if they're already written 🙂

Reply:
The difference is fairly slim, but I'm happy to keep these if you are.

We've quite an inconsistent approach to VRT and testing in general - some widgets/components have none, this has the full gamut on the VRT side. I'd prefer to see more in general, even if it means a slower test run. Anyway that's a tangential issue, one for another time!